File: tests/uasm/sample/string_switch_default/StringSwitchDefaultTest.cs
using UdonSharp;
using UnityEngine;

[UdonBehaviourSyncMode(BehaviourSyncMode.None)]
public class StringSwitchDefaultTest : UdonSharpBehaviour
{
    public void Start()
    {
        string command = "pause";
        string status = "";

        switch (command)
        {
            case "start":
                status = "started";
                break;
            case "stop":
                status = "stopped";
                break;
            default:
                status = "unknown";
                break;
        }

        Debug.Log(status);
    }
}

File: tests/uasm/sample/string_switch_default/StringSwitchDefaultTest.ts
import { UdonBehaviour } from "@ootr/udon-assembly-ts/stubs/UdonDecorators";
import { UdonSharpBehaviour } from "@ootr/udon-assembly-ts/stubs/UdonSharpBehaviour";
import { Debug } from "@ootr/udon-assembly-ts/stubs/UnityTypes";

@UdonBehaviour()
export class StringSwitchDefaultTest extends UdonSharpBehaviour {
  Start(): void {
    const command: string = "pause";
    let status = "";

    switch (command) {
      case "start":
        status = "started";
        break;
      case "stop":
        status = "stopped";
        break;
      default:
        status = "unknown";
        break;
    }

    Debug.Log(status);
  }
}
